### ISSUE TITLE
False positive for CS8604 when using the logical XOR operator

### ISSUE THREAD

**Issue by CollinAlpert** [dotnet/roslyn#69080]
Nullable reference type heuristics are faulty when using the logical XOR operator (^).

**Version Used**: 4.7.0-3.23328.4 (7931076b)

**Steps to Reproduce**:

```csharp
int? value1 = null;
string? value2 = null;
if (!value1.HasValue ^ value2 is null)
{
    return;
}

if (value1.HasValue)
{
    Run(value2); // <-- CS8604: Possible null reference argument for parameter 's' in 'void Run(string s)'.
}

void Run(string s)
{
    System.Console.WriteLine(s);
}
```
https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwJZWAfgAQDcCGANhAKYCM2AvNlBIYQNwCwAUAAJkAMOBxJATFRp1GrVqgBm2ABQBCXqTIA6ABL4AzgDUipbAD08OgdlTrh9AJSsA3q2z3sbAOzMWAXzEtJMheVUbtPisWWxYHbAAlaGlffgtXDxZWNgAWSOiOTmx1YNDwjiUOAE5pHISgA

**Expected Behavior**: No warning is emitted, since if `value1` is not null, `value2` is guaranteed to not be null as well.

**Actual Behavior**: CS8604 is emitted at the `Run` invocation.

**Comment by RikkiGibson** [comment]
Thanks for reporting this issue. It looks like this is a duplicate of #36927.
